test/Microsoft.Extensions.Localization.Tests/ResourceManagerStringLocalizerTest.cs

```diff
@@ -233,49 +242,57 @@ private static int GetCultureInfoDepth(CultureInfo culture)
 
         public class TestResourceManager : ResourceManager
         {
-            public TestResourceManager(string baseName, Assembly assembly)
-                : base(baseName, assembly)
+            private AssemblyWrapper _assemblyWrapper;
+
+            public TestResourceManager(string baseName, AssemblyWrapper assemblyWrapper)
+                : base(baseName, assemblyWrapper.Assembly)
             {
+                _assemblyWrapper = assemblyWrapper;
             }
 
             public override string GetString(string name, CultureInfo culture) => null;
+
+            public override ResourceSet GetResourceSet(CultureInfo culture, bool createIfNotExists, bool tryParents)
+            {
+                var resourceStream = _assemblyWrapper.GetManifestResourceStream(BaseName);
+
+                return resourceStream != null ? new ResourceSet(resourceStream) : null;
+            }
         }
 
-        public class TestResourceStringProvider : AssemblyResourceStringProvider
+        public class TestResourceStringProvider : ResourceManagerStringProvider
         {
-            private TestAssemblyWrapper _assemblyWrapper;
-
             public TestResourceStringProvider(
                     IResourceNamesCache resourceCache,
-                    TestAssemblyWrapper assemblyWrapper,
+                    TestResourceManager resourceManager,
+                    Assembly assembly,
                     string resourceBaseName)
-                : base(resourceCache, assemblyWrapper, resourceBaseName)
-            {
-                _assemblyWrapper = assemblyWrapper;
-            }
-
-            protected override AssemblyWrapper GetAssembly(CultureInfo culture)
+                : base(resourceCache, resourceManager, assembly, resourceBaseName)
             {
-                return _assemblyWrapper;
             }
         }
 
         public class TestAssemblyWrapper : AssemblyWrapper
         {
-            public TestAssemblyWrapper(string name = nameof(TestAssemblyWrapper))
-                : base(typeof(TestAssemblyWrapper).GetTypeInfo().Assembly)
+            public TestAssemblyWrapper()
+                : this(typeof(TestAssemblyWrapper))
             {
-                FullName = name;
             }
 
-            public int GetManifestResourceStreamCallCount { get; private set; }
+            public TestAssemblyWrapper(Type type)
+                : base(type.GetTypeInfo().Assembly)
+            {
+            }
 
-            public override string FullName { get; }
+            public bool HasResources { get; set; } = true;
+
+            public int GetManifestResourceStreamCallCount { get; private set; }
```

Comment:
`ManifestResourceStreamCallCount` is better than `GetManifestResourceStreamCallCount`